rpm/fedora-39/Dockerfile

```diff
@@ -0,0 +1,25 @@
+# syntax=docker/dockerfile:1
+
+ARG GO_IMAGE
+ARG DISTRO=fedora
+ARG SUITE=39
+ARG BUILD_IMAGE=${DISTRO}:${SUITE}
+
+FROM ${GO_IMAGE} AS golang
+
+FROM ${BUILD_IMAGE}
+ENV GOPROXY=https://proxy.golang.org|direct
+ENV GO111MODULE=off
+ENV GOPATH /go
+ENV PATH $PATH:/usr/local/go/bin:$GOPATH/bin
+ENV AUTO_GOPATH 1
+ARG DISTRO
+ARG SUITE
+ENV DISTRO=${DISTRO}
+ENV SUITE=${SUITE}
+RUN dnf5 install -y rpm-build rpmlint dnf5-plugins
+COPY --link SPECS /root/rpmbuild/SPECS
+RUN dnf5 builddep -y /root/rpmbuild/SPECS/*.spec
```

Comment:
This PR uses `dnf5` because `dnf` was providing some conflict errors

```
file /etc/dnf/dnf.conf from install of dnf-data-4.16.2-4.fc40.noarch conflicts with file from package libdnf5-5.1.0-2.fc39.x86_64
```